common/src/main/scala/uk/ac/wellcome/models/transformable/miro/MiroTransformChecks.scala

```diff
@@ -0,0 +1,97 @@
+package uk.ac.wellcome.models.transformable.miro
+
+import uk.ac.wellcome.models.transformable.{
+  FieldIssues,
+  ShouldNotTransformException
+}
+
+trait MiroTransformChecks {
+  /* XML tags refer to fields within the Miro XML dumps.
+   * If the <image_cleared> or <image_copyright_cleared> fields are
+   * missing or don't have have the value 'Y', then we shouldn't expose
+   * this image in the public API.
+   *
+   * See https://github.com/wellcometrust/platform-api/issues/356
+   */
+  private def checkCopyrightCleared(
+    data: MiroTransformableData): List[FieldIssues] = {
+    val generalCleared = if (data.cleared.getOrElse("N") != "Y") {
+      List(FieldIssues("image_cleared", data.cleared))
+    } else Nil
+
+    val copyrightCleared = if (data.copyrightCleared.getOrElse("N") != "Y") {
+      List(FieldIssues("image_cleared", data.cleared))
+    } else Nil
+
+    generalCleared ++ copyrightCleared
+  }
+
+  /* There are a bunch of <image_tech_*> fields that refer to the
+   * underlying image file.  If these are empty, there isn't actually a
+   * file to retrieve, which breaks the Collection site.  Sometimes this is
+   * a "glue" record that refers to multiple images.  e.g. V0011212ETL
+   *
+   * Eventually it might be nice to collate these -- have all the images
+   * in the same API result, but for now, we just exclude them from
+   * the API.  They aren't useful for testing image search.
+   */
+  private def checkImageExists(
+    data: MiroTransformableData): List[FieldIssues] =
+    if (data.techFileSize.getOrElse(List[String]()).isEmpty) {
+      List(FieldIssues(
+        "image_tech_file_size",
+        data.techFileSize,
+        Some(
+          "Missing image_tech_file_size means there is no underlying image")))
```

Comment:
While we’re here, I might couch this statement slightly:

> Missing image_tech_file_size means there is *likely* no underlying image

The only way to actually know is ask S3 for the JPEG derivative, but that's expensive.